nixos/modules/system/activation/bootspec.nix

```diff
@@ -0,0 +1,111 @@
+# Note that these schemas are defined by RFC-0125.
+# This document is considered a stable API, and is depended upon by external tooling.
+# Changes to the structure of the document, or the semantics of the values should go through an RFC.
+#
+# See: https://github.com/NixOS/rfcs/pull/125
+{ config
+, pkgs
+, lib
+, ...
+}:
+let
+  cfg = config.boot.bootspec;
+  children = lib.mapAttrs (childName: childConfig: childConfig.configuration.system.build.toplevel) config.specialisation;
+  schemas = {
+    v1 = rec {
+      filename = "boot.json";
+      json =
+        pkgs.writeText filename
+          (builtins.toJSON
+          {
+            v1 = {
+              kernel = "${config.boot.kernelPackages.kernel}/${config.system.boot.loader.kernelFile}";
+              kernelParams = config.boot.kernelParams;
+              initrd = "${config.system.build.initialRamdisk}/${config.system.boot.loader.initrdFile}";
+              initrdSecrets = "${config.system.build.initialRamdiskSecretAppender}/bin/append-initrd-secrets";
+              label = "NixOS ${config.system.nixos.codeName} ${config.system.nixos.label} (Linux ${config.boot.kernelPackages.kernel.modDirVersion})";
+
+              inherit (cfg) extensions;
+            };
+          });
+
+      generator =
+        let
+          specialisationLoader = (lib.mapAttrsToList
+            (childName: childToplevel: lib.escapeShellArgs [ "--slurpfile" childName "${childToplevel}/bootspec/${filename}" ])
+            children);
+        in
+        ''
+          mkdir -p $out/bootspec
+
+          # Inject toplevel and init in the bootspec.
+          # This can be done only here because we *cannot* depend on $out, except
+          # by living in $out itself.
+          ${pkgs.jq}/bin/jq '
+            .v1.toplevel = $toplevel |
+            .v1.init = $init
+            ' \
+            --sort-keys \
+            --arg toplevel "$out" \
+            --arg init "$out/init" \
+            < ${json} \
+            | ${pkgs.jq}/bin/jq \
+              --sort-keys \ # Slurp all specialisations and inject them as values in .specialisations.{name} = {specialisation bootspec}.
```

Comment:
The comment causes the following build error:

```
jq: error: Could not open file Slurp: No such file or directory
jq: error: Could not open file all: No such file or directory
jq: error: Could not open file specialisations: No such file or directory
jq: error: Could not open file and: No such file or directory
jq: error: Could not open file inject: No such file or directory
jq: error: Could not open file them: No such file or directory
jq: error: Could not open file as: No such file or directory
jq: error: Could not open file values: No such file or directory
jq: error: Could not open file in: No such file or directory
jq: error: Could not open file .specialisations.{name}: No such file or directory
jq: error: Could not open file =: No such file or directory
jq: error: Could not open file {specialisation: No such file or directory
jq: error: Could not open file bootspec}.: No such file or directory
```

Reply:
Ugh, thank you !